virtualizarr/writers/icechunk.py

```diff
@@ -124,15 +134,37 @@ def write_virtual_variable_to_icechunk(
     group: "Group",
     name: str,
     var: Variable,
+    append_dim: Optional[str] = None,
 ) -> None:
     """Write a single virtual variable into an icechunk store"""
     ma = cast(ManifestArray, var.data)
     zarray = ma.zarray
+    shape = zarray.shape
+    mode = store.mode.str
+
+    # Aimee: resize the array if it already exists
+    # TODO: assert chunking and encoding is the same
+    existing_keys = tuple(group.array_keys())
+    append_axis, existing_num_chunks = None, None
+    if name in existing_keys and mode == "a":
+        # resize
+        dims = var.dims
+        if append_dim in dims:
+            append_axis = dims.index(append_dim)
+            existing_array = group[name]
+            existing_size = existing_array.shape[append_axis]
+            existing_num_chunks = int(
+                existing_size / existing_array.chunks[append_axis]
+            )
+            new_shape = list(existing_array.shape)
+            new_shape[append_axis] += var.shape[append_axis]
+            shape = tuple(new_shape)
```

Comment:
I wonder if some axis-handling logic from the `concat` function I wrote for `ManifestArray` could be re-used here.

Reply:
Looking at https://github.com/zarr-developers/VirtualiZarr/blob/main/virtualizarr/manifests/array_api.py#L88 I think some of that could definitely be reused here.